packages/template-retail-react-app/app/main.jsx

```diff
@@ -12,3 +12,13 @@ const main = () => {
 }
 
 main()
+
+// since v2 storefront uses localStorage.setItem, which does not trigger eventListener set
+// from preview script which in charge of communication between RA and storefront.
+// we need this customisation to make sure it can work with Storefront Preview feature
+window.onload = () => {
+    window.STOREFRONT_PREVIEW = {
+        ...window.STOREFRONT_PREVIEW,
+        getToken: () => window.localStorage.getItem('token')
+    }
+}
```

Comment:
```suggestion
```
❗ Now that we poll for changes, rather than listen for events, the default getToken implementation works for v2 without requiring customization.

Reply:
Wait, I thought we were going to switch to polling only, but we seem to have kept the dual polling/events logic. This would be necessary, then.

Reply:
default getToken implementation for V2 won't work, we need to use polling for V2.